libs/libc/unistd/lib_getopt.c

```diff
@@ -128,11 +134,11 @@ int getopt(int argc, FAR char * const argv[], FAR const char *optstring)
        * the program, optind must be reset to some value <= 1.
        */
 
-      if (optind < 1 || !g_binitialized)
+      if (go->go_optind < 1 || !g_binitialized)
         {
-          optarg         = NULL;
-          optind         = 1;     /* Skip over the program name */
-          optopt         = '?';
+          go->go_optarg  = NULL;
+          go->go_optind  = 1;     /* Skip over the program name */
+          go->go_optopt  = '?';
           g_optptr       = NULL;  /* Start at the beginning of the first argument */
           g_binitialized = true;  /* Now we are initialized */
```

Comment:
`g_optptr`, `g_argv`, `g_argc`, and `g_binitialized` also need to be in TLS.

Reply:
Done!  Thanks for your good eyes!